tests/ocp/sriov/internal/ocpsriovconfig/config.go

```diff
@@ -6,24 +7,84 @@ import (
+	// DefaultVFNum is the default number of virtual functions to create.
+	DefaultVFNum = 2
 )
 
+// DeviceConfig represents a SR-IOV device configuration.
+type DeviceConfig struct {
+	Name              string `yaml:"name"`
+	DeviceID          string `yaml:"device_id"`
+	Vendor            string `yaml:"vendor"`
+	InterfaceName     string `yaml:"interface_name"`
+	SupportsMinTxRate bool   `yaml:"supports_min_tx_rate"`
+}
+
 // SriovOcpConfig type keeps sriov configuration.
 type SriovOcpConfig struct {
 	*ocpconfig.OcpConfig
-	OcpSriovOperatorNamespace string `yaml:"sriov_operator_namespace" envconfig:"ECO_OCP_SRIOV_OPERATOR_NAMESPACE"`
-	OcpSriovTestContainer     string `yaml:"ocp_sriov_test_container" envconfig:"ECO_OCP_SRIOV_TEST_CONTAINER"`
+	OcpSriovOperatorNamespace string         `yaml:"sriov_operator_namespace" envconfig:"ECO_OCP_SRIOV_OPERATOR_NAMESPACE"`
+	OcpSriovTestContainer     string         `yaml:"ocp_sriov_test_container" envconfig:"ECO_OCP_SRIOV_TEST_CONTAINER"`
+	SriovInterfaces           string         `envconfig:"ECO_OCP_SRIOV_INTERFACE_LIST"`
+	Devices                   []DeviceConfig `yaml:"devices"`
+	VFNum                     int            `yaml:"vf_num" envconfig:"ECO_OCP_SRIOV_VF_NUM"`
+}
+
+// sriovYAMLConfig and sriovEnvConfig are temporary structs used to decode configuration
+// without affecting the embedded OcpConfig/GeneralConfig structs.
+//
+// Why we need these separate structs:
+// SriovOcpConfig embeds *ocpconfig.OcpConfig which itself embeds *generalconfig.GeneralConfig.
+// When yaml.v2 decodes into a struct with embedded pointer fields, it resets those pointers
+// to nil for any fields not present in the YAML file - even if they were already initialized.
+//
+// For example, if GeneralConfig.WorkerLabel was set to "node-role.kubernetes.io/worker"
+// by the parent config, decoding a YAML file that only contains "sriov_operator_namespace"
+// would reset WorkerLabel to "" because yaml.v2 zeroes out the embedded struct first.
+//
+// By decoding into a separate struct that only contains sriov-specific fields, we avoid
+// this issue and can selectively copy values to the main config without losing parent values.
+//
+// The envconfig library has similar behavior, so we use the same pattern for environment
+// variable processing.
+type sriovYAMLConfig struct {
+	OcpSriovOperatorNamespace string         `yaml:"sriov_operator_namespace"`
+	OcpSriovTestContainer     string         `yaml:"ocp_sriov_test_container"`
+	Devices                   []DeviceConfig `yaml:"devices"`
+	VFNum                     int            `yaml:"vf_num"`
+}
+
+type sriovEnvConfig struct {
+	OcpSriovOperatorNamespace string `envconfig:"ECO_OCP_SRIOV_OPERATOR_NAMESPACE"`
+	OcpSriovTestContainer     string `envconfig:"ECO_OCP_SRIOV_TEST_CONTAINER"`
 	SriovInterfaces           string `envconfig:"ECO_OCP_SRIOV_INTERFACE_LIST"`
+	VFNum                     int    `envconfig:"ECO_OCP_SRIOV_VF_NUM"`
+}
```

Comment:
I beleive we don't need it. Please let me know if you need any help with that. As an example you can use:
https://github.com/rh-ecosystem-edge/eco-gotests/blob/main/tests/cnf/core/network/internal/netconfig/config.go